infobip-typescript-generator-extension-api/src/main/java/com/infobip/typescript/transformer/ClassTransformerDecoratorExtension.java

```diff
@@ -59,29 +57,50 @@ private TsBeanModel decorateClass(SymbolTable symbolTable, TsBeanModel bean) {
     }
 
     private TsPropertyModel getDecorators(SymbolTable symbolTable, TsBeanModel model, TsPropertyModel tsPropertyModel) {
-        return getField(model, tsPropertyModel).map(Field::getType)
-                                               .filter(type -> Modifier.isAbstract(
-                                                       type.getModifiers()) || type.isInterface())
-                                               .map(type -> tsPropertyModel.withDecorators(
-                                                       getDecorators(symbolTable, model, tsPropertyModel, type)))
+        return getField(model, tsPropertyModel).map(this::getParameterizedTypeClasses)
+                                               .map(parameterizedTypeClasses -> tsPropertyModel.withDecorators(
+                                                       getDecorators(symbolTable, model, tsPropertyModel,
+                                                                     resolveTypeToDecorate(parameterizedTypeClasses))))
                                                .orElse(tsPropertyModel);
     }
 
+    private Class<?> resolveTypeToDecorate(ParameterizedTypeClasses parameterizedTypeClasses) {
+        Class<?> type = parameterizedTypeClasses.getType();
+        Class<?> typeToDecorate = type;
+        if (type.isArray()) {
+            typeToDecorate = type.getComponentType();
+        }
+
+        Optional<Class<?>> typeArgument = parameterizedTypeClasses.getTypeArgument();
+        if (Collection.class.isAssignableFrom(type) && typeArgument.isPresent()) {
+            typeToDecorate = typeArgument.get();
+        }
+        return typeToDecorate;
+    }
+
     private List<TsDecorator> getDecorators(SymbolTable symbolTable,
                                             TsBeanModel model,
                                             TsPropertyModel tsPropertyModel,
                                             Class<?> type) {
         return factory.create(type)
-                      .filter(resolver -> resolver instanceof CompositeJsonTypeResolver<?>)
+                      .filter(resolver -> isHierarchicalDecoratorNeeded(resolver, type))
```

Comment:
oh i get it, you have reverse logic, instead of finding leaves you have detected root :) nice, i think that we have now better use case coverage :)

Reply:
Yes, and the checks in `isHierarchicalDecoratorNeeded` and `isHierarchyRoot` are the same ones that we had before, so this fix one breakage that my previous changes had introduced in the generates Single Send typescript